pyccel/ast/core.py

```diff
@@ -3861,6 +3810,90 @@ def __str__(self):
                 args   = ', '.join(self.args),
                 result = result)
 
+class Interface(Basic):
+
+    """Represents an Interface.
+
+    Parameters
+    ----------
+    name : str
+        The name of the interface.
+
+    functions : iterable
+        The functions of the interface.
+
+    hide: bool
+        if True, the interface definition will not be generated.
+
+    is_argument: bool
+        True if the interface is used for a function argument.
+
+    Examples
+    --------
+    >>> from pyccel.ast.core import Interface, FunctionDef
+    >>> f = FunctionDef('F', [], [], [])
+    >>> Interface('I', [f])
+    """
+
+    def __new__( cls, *args, **kwargs ):
+        return Basic.__new__(cls)
+
+    def __init__(
+        self,
+        name,
+        functions,
+        hide=False,
+        is_argument = False,
+        ):
+
+        if not isinstance(name, str):
+            raise TypeError('Expecting an str')
+        if not isinstance(functions, list):
+            raise TypeError('Expecting a list')
+        self._name = name
+        self._functions = functions
+        self._hide = hide
```

Comment:
Is there any point keeping this variable if you removed the access function?

Reply:
I forgot it, I will remove it, thanks

Reply:
Don't forget the docstring

Reply:
Yes I removed it too